### ISSUE TITLE
New Legends: slowdowns

### ISSUE THREAD

**Issue by HadetTheUndying** [xemu-project/xemu#1007]
### Title

https://xemu.app/titles/54510001/#New-Legends

### Bug Description

First Mission Starts and Runs very inconsistently. Frequent huge slowdowns from the game's 60FPS target.

I did some performance profiling in `hotspot` and discovered what appears to eating up about half the cycle time when it's happening.
![nicememe tr6wEOQWj8FSnByc4Kf](https://user-images.githubusercontent.com/28496334/171497786-d86f1238-48cc-4109-b420-09cd5ddeb17d.png)

https://user-images.githubusercontent.com/28496334/171498213-0e4314cf-ea9c-4e72-91d1-19aeddc638aa.mp4

### Expected Behavior

Performance to remain as *stable* as it is on hardware.

### xemu Version

Version:      0.7.25
Branch:       master
Commit:       7d6da227ba3b224647e35d2385419a3bec25a3b3
Date:         Tue May 31 08:20:36 PM UTC 2022

### System Information

CPU:          AMD Ryzen 7 2700X Eight-Core Processor         
OS Platform:  Linux
OS Version:   Void Linux
Manufacturer: AMD
GPU Model:    AMD Radeon RX Vega (vega10, LLVM 12.0.1, DRM 3.44, 5.16.20_1)
Driver:       4.6 (Core Profile) Mesa 22.2.0-devel (git-982cc9bcf5)
Shader:       4.60

### Additional Context

I don't expect a lot of other reports, when I went to the game store to pick this game up the guy at the counter asked me twice if I was sure I wanted to buy it. This probably does effect other games, but I couldn't find a similar issue searching current issues.

**Comment by abaire** [comment]
Maybe I'm missing something from the screenshot; did you find a smoking gun entry that accounts for the time (i.e., has a high "cycles self")?

It'd be expected that a lot of time would be spent in the BEGIN_END handler since that's what does the vast majority of the actual rendering in a typical game. From that trace it seems to be doing a shader compilation, if it's really recompiling them super often it might be worth testing the shader caching PR to see what effect it has.

**Comment by HadetTheUndying** [comment]
> Maybe I'm missing something from the screenshot; did you find a smoking gun entry that accounts for the time (i.e., has a high "cycles self")?
> 
> It'd be expected that a lot of time would be spent in the BEGIN_END handler since that's what does the vast majority of the actual rendering in a typical game. From that trace it seems to be doing a shader compilation, if it's really recompiling them super often it might be worth testing the shader caching PR to see what effect it has.

If you wanna help me figure it out later that'd be good. i was just sorting by cycle time. I'll test with the Shader Caching PR though

**Comment by HadetTheUndying** [comment]
Yes https://github.com/mborgerson/xemu/pull/768 fixes the performance.

**Comment by Triticum0** [comment]
Retested on my machine performance is better that what on issue. not perfect though
